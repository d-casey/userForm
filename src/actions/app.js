import {
  RESET_FORM_STAGE,
  UPDATE_STAGE
} from '../constants/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})
