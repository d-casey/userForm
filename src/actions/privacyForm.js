import {
  SUBMIT_PRIVACY_FORM,
  RESET_PRIVACY_FORM
} from '../constants/actions'

export const submitPrivacyForm = formFields => ({
  type: SUBMIT_PRIVACY_FORM,
  payload: formFields
})

export const resetPrivacyForm = () => ({
  type: RESET_PRIVACY_FORM
})
