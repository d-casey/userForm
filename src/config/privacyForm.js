const privacyForm = {
  checkboxData: [
    {
      name: 'receive_updates',
      defaultChecked: true,
      message: 'Receive Updates about Tray.io products by email',
      testId: 'receive_updates'
    },
    {
      name: 'receive_communication',
      message: 'Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team Receive communication by email for other products created by the Tray.io team',
      testId: 'receive_communication'
    }
  ]
}

export default privacyForm
