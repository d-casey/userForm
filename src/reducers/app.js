import {
  UPDATE_STAGE,
  RESET_FORM_STAGE
} from '../constants/actions'

const initialState = {
  stage: 0
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_STAGE:
      return {
        ...state,
        stage: action.payload.stage,
      }

      case RESET_FORM_STAGE:
        return {
          ...state,
          stage: 0
        }

    default:
      return state
  }
}
