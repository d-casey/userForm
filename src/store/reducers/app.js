import {
  UPDATE_STAGE,
  RESET_FORM_STAGE,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

const initialState = {
  stage: 0,
  formCompleted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_STAGE:
      return {
        ...state,
        stage: action.payload.stage,
      }

    case RESET_FORM_STAGE:
      return {
        ...state,
        stage: 0
      }

    case SET_FORM_COMPLETION:
      return {
        ...state,
        formCompleted: true
      }

    case RESET_APP:
      return {
        ...state,
        formCompleted: false,
        stage: 0
      }

    default:
      return state
  }
}
