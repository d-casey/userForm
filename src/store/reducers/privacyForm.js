import {
  RESET_PRIVACY_FORM,
  SUBMIT_PRIVACY_FORM
} from '../constants/actions'

const initialState = {
  receive_updates: true,
  receive_communication: false,
  isSubmitted: false
}

export default (state = initialState, action) => {
  switch (action.type) {
    case SUBMIT_PRIVACY_FORM:
      return {
        ...state,
        receive_updates: action.payload.receive_updates,
        receive_communication: action.payload.receive_communication,
        isSubmitted: true
      }

      case RESET_PRIVACY_FORM:
        return initialState

    default:
      return state
  }
}
