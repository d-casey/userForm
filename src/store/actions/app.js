import {
  RESET_FORM_STAGE,
  UPDATE_STAGE,
  LOGGING_DETAILS,
  SET_FORM_COMPLETION,
  RESET_APP
} from '../constants/actions'

import { resetUserForm, resetPrivacyForm } from '../../store/actions'

export const updateStage = stage => ({
  type: UPDATE_STAGE,
  payload: {
    stage
  }
})

export const resetFormStage = () => ({
  type: RESET_FORM_STAGE
})

export const resetAllForms = () => dispatch => {
  dispatch(resetUserForm())
  dispatch(resetPrivacyForm())
}

export const setFormCompletion = () => ({
  type: SET_FORM_COMPLETION
})

export const resetApp = () => ({
  type: RESET_APP
})

export const loggingDetails = () => ({
  type: LOGGING_DETAILS
})

export const logSubmittedFormDetails = (formDetails) => dispatch => {
  dispatch(loggingDetails())
  console.log('formDetails: ', formDetails)
  dispatch(resetAllForms())
  dispatch(setFormCompletion())
}
