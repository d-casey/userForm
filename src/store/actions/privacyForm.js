import {
  SUBMIT_PRIVACY_FORM,
  RESET_PRIVACY_FORM
} from '../constants/actions'

import { updateStage } from '../actions'

export const updatePrivacyFields = formFields => ({
  type: SUBMIT_PRIVACY_FORM,
  payload: formFields
})

export const submitPrivacyForm = formFields => dispatch => {
  dispatch(updatePrivacyFields(formFields))
  dispatch(updateStage(2))
}

export const resetPrivacyForm = () => ({
  type: RESET_PRIVACY_FORM
})
