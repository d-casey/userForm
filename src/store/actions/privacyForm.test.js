import * as appActions from './app'
import * as privacyFormActions from './privacyForm'

describe('privacyForm actions', () => {
  describe('updatePrivacyFields', () => {
    expect(privacyFormActions.updatePrivacyFields({ formFields: {} })).toMatchSnapshot()
  })

  describe('resetPrivacyForm', () => {
    it('sends the resetPrivacyForm action', () => {
      expect(privacyFormActions.resetPrivacyForm()).toMatchSnapshot()
    })
  })

  describe('submitPrivacyForm', () => {
    let dispatch, formFields
    beforeEach(() => {
      dispatch = jest.fn()
      formFields = 'FORM'
      privacyFormActions.submitPrivacyForm(formFields)(dispatch)
    })

    test('it calls dispatch two times', () => {
      expect(dispatch).toHaveBeenCalledTimes(2)
    })

    it('calls dispatch with the updatePrivacyFields action', () => {
      expect(dispatch).toBeCalledWith({'payload': 'FORM', 'type': 'SUBMIT_PRIVACY_FORM'})
    })

    it('calls dispatch to update the stage', () => {
      expect(dispatch).toBeCalledWith({'payload': {'stage': 2}, 'type': 'UPDATE_STAGE'})
    })
  })
})
