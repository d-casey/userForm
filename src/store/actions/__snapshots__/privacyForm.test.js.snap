// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "payload": Object {
    "formFields": Object {},
  },
  "type": "SUBMIT_PRIVACY_FORM",
}
`;

exports[`privacyForm actions resetPrivacyForm sends the resetPrivacyForm action 1`] = `
Object {
  "type": "RESET_PRIVACY_FORM",
}
`;
